plugin/metrics/prometheus/metricsstore/reader_test.go

```diff
@@ -64,42 +64,49 @@ var defaultConfig = config.Configuration{
 	LatencyUnit:                 "ms",
 }
 
-func tracerProvider(t *testing.T) (trace.TracerProvider, *tracetest.InMemoryExporter) {
+func tracerProvider(t *testing.T) (trace.TracerProvider, *tracetest.InMemoryExporter, func(ctx context.Context) error) {
 	exporter := tracetest.NewInMemoryExporter()
 	tp := sdktrace.NewTracerProvider(
 		sdktrace.WithSampler(sdktrace.AlwaysSample()),
 		sdktrace.WithSyncer(exporter),
 	)
-	return tp, exporter
+	closer := func(ctx context.Context) error {
```

Comment:
you don't need to return a closer, the callers can call tp.Shutdown themselves, you do not provide any extra capability by wrapping it.

Reply:
however, what you COULD do is this:
```
closer := func() {
		assert.NoError(t, tp.Shutdown(context.Background())
	}
```
This way you make it very simple for the caller to do `defer closer()`

Reply:
As I told, the return type is trace.TracerProvider which is not sdktrace. If I change it, `otel.Tp` won't support it.